### PR TITLE
Added kubebuilder:pruning:PreserveUnknownFields

### PR DESCRIPTION
**Pull request by respringer**

### CODE REVIEW

File: operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go [hunk 1]
```diff
@@ -79,6 +79,7 @@ type CassandraDatacenterSpec struct {
 	ServerType string `json:"serverType"`
 
 	// Config for the server, in YAML format
+	// +kubebuilder:pruning:PreserveUnknownFields
```
Comment (sandoichi) [inline operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go:82]: :100:

**Review by sandoichi — APPROVED** [review]

File: charts/cass-operator-chart/templates/customresourcedefinition.yaml [hunk 2]
```diff
@@ -6307,9 +6307,7 @@ spec:
               type: string
           type: object
       type: object
-      {{- if semverCompare ">= 1.15-0" .Capabilities.KubeVersion.GitVersion }}
```
Comment (jimdickinson) [inline charts/cass-operator-chart/templates/customresourcedefinition.yaml:6310]: I think we need to keep this if we want to support k8s <= 1.14 - which we might be OK moving away from

File: operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go (hunk 1 above)
Comment (jimdickinson) [reply operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go:82]: 🎉

File: charts/cass-operator-chart/templates/customresourcedefinition.yaml (hunk 2 above)
Comment (respringer) [reply charts/cass-operator-chart/templates/customresourcedefinition.yaml:6310]: @jimdickinson should I put the conditional back?

Comment (jimdickinson) [reply charts/cass-operator-chart/templates/customresourcedefinition.yaml:6310]: for this PR, I think so... I thought we had some more hackery around getting `x-kubernetes-preserve-unknown-fields: true` into the yaml, but maybe that had already been cleaned up and turned into "apply the patch"

**Pull request merged**